libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py

```diff
@@ -586,8 +604,14 @@ def get_available_models(
         **kwargs: Any,
     ) -> List[Model]:
         """Retrieve a list of available models."""
+
         available = [
-            model for model in self.client.available_models if model.client == filter
+            model
+            for model in self.client.available_models
+            if (
+                model.client == filter
+                or (model.base_model and model.base_model != model.id)
+            )
```

Comment:
what about something like this?

```
diff --git a/libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py b/libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py
index 5b1e7ae..3c04377 100644
--- a/libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py
+++ b/libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py
@@ -587,7 +587,7 @@ class _NVIDIAClient(BaseModel):
     ) -> List[Model]:
         """Retrieve a list of available models."""
         available = [
-            model for model in self.client.available_models if model.client == filter
+            model for model in self.client.available_models
         ]
 
         # if we're talking to a hosted endpoint, we mix in the known models
```

i think it's actually a bug that we filter out models when talking to a non-hosted nim.

Reply:
yes it is, but if we remove filter then I guess all models will get listed and the client based filtering will not happen like chat models for ChatNVIDIA client etc.

Reply:
what about continuing it with something like...
```
@@ -595,9 +595,7 @@ class _NVIDIAClient(BaseModel):
         # the NV-Embed-QA and VLM models are hosted on ai.api.nvidia.com
         # instead of integrate.api.nvidia.com.
         if self.is_hosted:
-            known = set(
-                model for model in MODEL_TABLE.values() if model.client == filter
-            )
-            available = list(set(available) | known)
+            known = set(MODEL_TABLE.values())
+            available = [model for model in set(available) | known if model.client == filter]
 
         return available
```

Reply:
yeah, makes sense to me